### ISSUE TITLE
The RandomDistributedScalarEncoder needs ability to set ```numBuckets``` 

### ISSUE THREAD

**Issue by cogmission** [numenta/htm.java#443]
```numBuckets``` needs an official Parameter in ```Parameters.java``` and it needs to be set by the Network API upon initialization. 

Care must be taken to guarantee that the ```resolution``` is properly calculated and that the param is tested when absent as well.

**Comment by rhyolight** [comment]
So will `Parameters` have access to the input data to calculate the min / max values required to calculate the `resolution`?

**Comment by cogmission** [comment]
I don't believe this is necessary. Alex mentioned that they are "auto-configured" from the ```numBuckets``` parameter... [as seen here](https://github.com/numenta/nupic/blob/master/src/nupic/encoders/random_distributed_scalar.py#L418) and [here](https://github.com/numenta/htm.java/blob/master/src/main/java/org/numenta/nupic/encoders/RandomDistributedScalarEncoder.java#L183)

I'm not averse to having encoder params at all. We do it for the DateEncoder (due to it's need for varied configuration), so we can do it for others as well?

**Comment by cogmission** [comment]
For instance, the NAB assumes that the ```numBuckets``` parameter is sufficient and doesn't specify ```resolution``` specifically, however HTM.Java doesn't have this so it thinks the ```resolution``` parameter is missing and throws java.lang.AnoyingAsHellException... :-P

My fix for this was to add ```resolution``` to the NAP JSON config file, but if we were going to do it the NuPIC way, we would allow the ```numBuckets``` parameter to be picked up and usher in the configuration.
